diff -Naur a/pkg.c b/pkg.c
--- a/pkg.c	2017-03-20 09:34:23.000000000 -0700
+++ b/pkg.c	2017-12-30 15:20:14.358635900 -0800
@@ -433,7 +433,7 @@
     Flag *flag = tmp->data;
     char *tmpstr = flag->arg;
 
-    if (pcsysrootdir != NULL && flag->type & (CFLAGS_I | LIBS_L)) {
+    if (pcsysrootdir != NULL && flag->type & (CFLAGS_I | LIBS_L) && strncmp(tmpstr+2, pcsysrootdir, strlen(pcsysrootdir)) != 0) {
       /* Handle non-I Cflags like -isystem */
       if (flag->type & CFLAGS_I && strncmp (tmpstr, "-I", 2) != 0) {
         char *space = strchr (tmpstr, ' ');
@@ -443,6 +443,9 @@
         g_string_append_len (str, tmpstr, space - tmpstr + 1);
         g_string_append (str, pcsysrootdir);
         g_string_append (str, space + 1);
+      } else if (pcsysrootdir != NULL && strncmp(tmpstr, "/usr", 4) == 0 && strncmp(tmpstr, pcsysrootdir, strlen(pcsysrootdir)) != 0) {
+        g_string_append (str, pcsysrootdir);
+        g_string_append (str, tmpstr);
       } else {
         g_string_append_c (str, '-');
         g_string_append_c (str, tmpstr[1]);
